dataactbroker/README.md

```diff
@@ -773,6 +773,153 @@ Example output if there are no files available:
 }
 ```
 
+# File Generation Routes
+
+## Generate Files
+**Route:** `/generate_file`
+
+**Method:** `POST`
+
+This route sends a request to the backend to utilize the relevant external APIs and generate the relevant file for the metadata that is submitted.
+
+**Deprecation Notice:** This route replaces `/generate_d1_file` and `/generate_d2_file`.
+
+### Body (JSON)
+
+```
+{
+    "submission_id": 123,
+    "file_type": "D1"
+    "start": "01/01/2016",
+    "end": "03/31/2016"
+}
+```
+
+### Body Description
+
+* `submission_id` - **required** - an integer representing the ID of the current submission
+* `file_type` - **required** - a string indicating the file type to generate. Allowable values are:
+	* `D1` - generate a D1 file
+	* `D2` - generate a D2 file
+	* `E` - generate a E file
+	* `F` - generate a F file
+* `start` - **required for D1/D2 only** - the start date of the requested period, in `MM/DD/YYYY` string format
+* `end` - **required for D1/D2 only** - the end date of the requested period, in `MM/DD/YYYY` string format
+
+### Response (JSON)
+Response will be the same format as those which are returned in the `/check_file` or `/check_d2_file` endpoint
+
+
+## File Status
+**Route:** `/check_generation_status`
+
+**Method:** `POST`
+
+This route returns either a signed S3 URL to the generated file or, if the file is not yet ready or have failed to generate for other reasons, returns a status indicating that.
+
+**Deprecation Notice:** This route replaces `/check_d1_file` and `/check_d2_file`.
+
+### Body (JSON)
+
+```
+{
+    "submission_id": 123,
+    "file_type": "D1"
+}
+```
+
+### Body Description
+
+* `submission_id` - An integer representing the ID of the current submission
+* `file_type` - **required** - a string indicating the file type whose status we are checking. Allowable values are:
+	* `D1` - generate a D1 file
+	* `D2` - generate a D2 file
+	* `E` - generate a E file
+	* `F` - generate a F file
+
+
+### Response (JSON)
+
+*State:* The file has successfully generated
+
+```
+{
+	"status": "finished",
+	"file_type": "D1",
+	"url": "https://........",
+	"start": "01/01/2016",
+	"end": "03/31/2016",
+	"message": ""
+}
+```
+
+*State:* The file is not yet ready
+
+```
+{
+	"status": "waiting",
+	"file_type": "D1",
+	"url": "",
+    "start": "01/01/2016",
+    "end": "03/31/2016",
+    "message": ""
+}
+```
+
+*State:* File generation has failed
+
+```
+{
+	"status": "failed",
+	"file_type": "D1",
+	"url": "",
+    "start": "01/01/2016",
+    "end": "03/31/2016",
+	"message": "The server could not reach the Federal Procurement Data System. Try again later."
+}
+```
+
+*State:* No file generation request has been made for this submission ID before
```

Comment:
Right, this is the case where /check_generations_status/ is called before /generate_file/